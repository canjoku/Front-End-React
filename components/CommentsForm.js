import { Component } from 'react'

class CommentsForm extends Component {
  constructor (props) {
    super(props)
    this.submitComment = this.submitComment.bind(this)
  }


  submitComment (e, mutation) {
    e.preventDefault()
    const form = e.target
    mutation({
      variables: {
        comment : {
          name: form.name.value,
          picture: form.picture.value,
          body: form.body.value,
          postSlug: this.props.slug
        }
      }
    })
    form.reset()
  }

  render () {
    const { mutation } = this.props
    return (
      <form onSubmit={(e) => this.submitComment(e, mutation)}>
        <input type="hidden" name="name" value={localStorage.getItem('user_name')}  />
        <input type= "hidden" name="picture" value={localStorage.getItem('user_picture')} />
        <input placeholder='Enter comment here' name="body" />
        <button>Submit</button>
      </form>
    )
  }
}

export default CommentsForm