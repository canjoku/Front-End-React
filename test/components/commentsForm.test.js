import CommentsForm from '../../components/CommentsForm'

test('<CommentsForm /> renders correctly', () => {
  const component = shallow(<CommentsForm />)
  expect(component).toMatchSnapshot()
})
