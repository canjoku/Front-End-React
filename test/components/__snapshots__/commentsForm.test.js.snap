// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentsForm /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentsForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="name"
          type="hidden"
          value={null}
        />,
        <input
          name="picture"
          type="hidden"
          value={null}
        />,
        <input
          name="body"
          placeholder="Enter comment here"
        />,
        <button>
          Submit
        </button>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "name",
          "type": "hidden",
          "value": null,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "picture",
          "type": "hidden",
          "value": null,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "body",
          "placeholder": "Enter comment here",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="name"
            type="hidden"
            value={null}
          />,
          <input
            name="picture"
            type="hidden"
            value={null}
          />,
          <input
            name="body"
            placeholder="Enter comment here"
          />,
          <button>
            Submit
          </button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "type": "hidden",
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "picture",
            "type": "hidden",
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "body",
            "placeholder": "Enter comment here",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
